### PR TITLE
Add support ConfigurationManager in Net Standard for configurable retry logic

### PR DESCRIPTION
**Pull request by DavoudEshtehari**
- Configuration retry logic functionality is fully supported by all target frameworks.
- Combine  `SqlConfigurableRetryLogicManager.cs` and `SqlConfigurableRetryLogicManager.CreateProvider.cs` and extract the `SqlConfigurableRetryLogicLoader.cs`.
- Improve resolving the app context switch value.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 1]
```diff
@@ -310,6 +310,14 @@
     <Compile Include="..\..\src\Microsoft\Data\SqlClient\Reliability\SqlConfigurableRetryLogicManager.cs">
       <Link>Microsoft\Data\SqlClient\Reliability\SqlConfigurableRetryLogicManager.cs</Link>
     </Compile>
+    <Compile Include="..\..\src\Microsoft\Data\SqlClient\Reliability\SqlConfigurableRetryLogicManager.CreateProvider.cs">
+      <Link>Microsoft\Data\SqlClient\Reliability\SqlConfigurableRetryLogicManager.CreateProvider.cs</Link>
+    </Compile>
+    <Compile Include="..\..\src\Microsoft\Data\SqlClient\Reliability\AppConfigManager.cs">
+      <Link>Microsoft\Data\SqlClient\Reliability\AppConfigManager.cs</Link>
+    </Compile>
+    <Compile Include="Microsoft\Data\SqlClient\Reliability\SqlConfigurableRetryLogicManager.NetCoreApp.cs" />
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:319]: Could we merge it with `SqlConfigurableRetryLogicManager.cs` above?

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 2]
```diff
@@ -310,6 +310,14 @@
...
+    <Compile Include="Microsoft\Data\SqlClient\Reliability\SqlAppContextSwitchManager.NetCoreApp.cs" />
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:320]: Also can we move the switch "EnableRetryLogicSwitch" from SqlRetryLogicProvider class to `LocalAppContextSwitches.cs` and load isRetryEnabled from there when needed statically? I'm not sure if it was intended to be non-static on each connection/command.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj (hunk 1 above)
Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:319]: `SqlConfigurableRetryLogicManager.cs` is used for both projects.

Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:319]: I can combine it with `SqlConfigurableRetryLogicManager.CreateProvider.cs` and extract `SqlConfigurableRetryLogicLoader` from these files with some changes in their names.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj (hunk 2 above)
Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:320]: I agree. I don't believe it was intended to be checked on every connection/command.

**Review by cheenamalhotra — APPROVED** [review]

**Review by johnnypham — APPROVED** [review]

**Pull request merged**